### DISCUSSION TITLE
Looking for Arturia KeyLab/MiniLab owners to improve compatiblity

### DISCUSSION THREAD

**Discussion by probonopd** [probonopd/MiniDexed#776] [General]
Hello. I have noticed that Arturia KeyLab/MiniLab controllers seem to be rather popular in the MiniDexed community.
When used with the bundled AnalogLab software, they can be used to control the software from the built-in screen, encoder, and knobs.

Unfortunately documentation about how this works is sparse and scattered. So in order to be able to support this in MiniDexed, I would like to have a mock (simulated) Arturia-compatible homebrew controller, that can simulate various Arturia KeyLab/MiniLab controllers and help in testing and debugging.

To make matters worse, I don't have any of these devices and from studying the available sources it seems that the different models send different MIDI commands.

So I am looking for volunteers with Arturia KeyLab/MiniLab controllers, e.g.,

* Arturia KeyLab ("mkI"): @simonvpe
* Arturia KeyLab mkII: ?
* Arturia KeyLab Essential: @magnusg-sa, @bearwolf, @warpeggio
* Arturia MiniLab 3: @soyersoyer, @GertSchepens

who could do some tests and find out which MIDI commands the controllers send in which situations.

Specifically, I would like to know the information I am collecting in this table:
https://github.com/probonopd/mock-arturia-controller/wiki (please edit that page directly).

It would be great if you could help me by filling in the values for your controller, and check anything marked "(tbc") = "to be confirmed".

I don't know whether you can activate the "AnalogLab" mode without running the bundled AnalogLab software. If that is possible, then it should be a matter of opening the DAW, pressing the buttons listed in the table, and watching the MIDI messages as they come in.
In case you need to be running AnalogLab, we need to find a way to intercept the MIDI messages sent from the controller to AnalogLab (e.g., using Wireshark to listen on USB).

Thank you very much!

**Comment by mgedda** [comment]
Hello, @magnusg-sa here (that is my work account which I used by accident in the MiniDexed communication, @mgedda is my private account). I checked the values on my KeyLab Essential 61 and they had to be updated.

I used the AnalogLab software in Ableton and registered the midi values with ["MIDI Monitor Plus" Max for Live plugin](https://maxforlive.com/library/device/7174/midi-monitor-plus). That could be a recommendation to other Ableton users.

For some reason I cannot push my changes (tried both with a new branch, and with master). So I provide my values here:

```
### While "Preset" is lit up (when browsing through presets)

| Short Name | <- button | -> button | Category | Preset | Encoder counter-clockwise | Encoder clockwise | Encoder click | Comments |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
...
| KeyLab Essential 61 | CC 28 value 127 | CC 29 value 127 | CC 116 value 127 | CC 117 value 127 | CC 114 value 64+63 (same when pressed down) | CC 114 value 64+65 (same when pressed down) | CC 115 value 127 |  |
```

```
### While "Category" is lit up (while navigating through the menu)

| Short Name |  <- button | -> button | Category | Preset | Encoder counter-clockwise | Encoder clockwise | Encoder click | Comments |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
...
| KeyLab Essential 61 | CC 28 value 127 | CC 29 value 127 | CC 116 value 127 | CC 117 value 127 | CC 112 value 64+63 (same when pressed down) | CC 112 value 64+65 (same when pressed down) | CC 113 value 127 |  |
```

I hope this helps.

All the best,
Magnus

**Reply by probonopd** [reply]
Thank you very much Magnus. This is a tremendous help. I can confirm that with your information, the mocked KeyLab Essential 61 is now working properly with AnalogLab. This means we can also use it as a test tatget to test future MiniDexed versions against. Again, thank you so much, made my day!

The wiki had the wrong permissions set, it is now corrected and should be publicly editable.

**Reply by mgedda** [reply]
I am happy I could be of help. Good luck with the project.

**Reply by probonopd** [reply]
If you don't mind, I have some additional question: 

In AnalogLab,
* What happens if you are in Category mode and press the Category button again? 
* Likewise, what happens if you are in Preset mode and press the Preset button again? 
* If you make a selection (e.g., "Piano) in Category mode and then switch to Preset mode, then switch to Category mode again, what do you see? Does it go back to the start of the menu (`<Instruments> Acid`) again, or does it remember where you were in the menu?
* In nested menus, is there a way to go "up"? E.g, Types -> Bass -> Bass Line. Is there a way to get bac up to "Instruments"?
* Bonus question (might be tougher to find out): What does it send in MCU (Mackie Control Universal) mode (that is used e.g., with Reaper)?

**Reply by mgedda** [reply]
Sorry, I don't have my setup available at the moment. I had to disassemble my setup after checking the midi values this morning. But I'll see if I can check later this week.
